### PR TITLE
OSSM-2082 CNI installer now creates the net.d directory if necessary (#638)

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Review by jwendell — APPROVED** [review]

File: cni/pkg/install/kubeconfig.go [hunk 1]
```diff
@@ -120,6 +120,11 @@ func createKubeconfigFile(cfg *config.InstallConfig, saToken string) (kubeconfig
 		return "", err
 	}
 
+	// When using Multus, the net.d dir might not exist yet, so we must create it
+	if err := os.MkdirAll(cfg.MountedCNINetDir, os.FileMode(0o755)); err != nil {
```
Comment (dgn) [inline cni/pkg/install/kubeconfig.go:124]: shouldn't this only return an error `if !os.IsExist(err)`?

Comment (luksa) [reply cni/pkg/install/kubeconfig.go:124]: From the docs: 

> If path is already a directory, MkdirAll does nothing and returns nil.

Comment (dgn) [reply cni/pkg/install/kubeconfig.go:124]: :facepalm: thanks

**Comment by tyxeron** [comment]
@luksa I am facing the same issue but with maistra-2.4.0.
```
Defaulted container "install-cni-v2-1" out of: install-cni-v2-1, install-cni-v2-2
2023-06-22T08:22:50.144857Z	info	Copied istio-cni to /host/opt/cni/bin/v2-1-istio-cni.
2023-06-22T08:22:50.415851Z	info	Copied istio-iptables to /host/opt/cni/bin/v2-1-istio-iptables.
2023-06-22T08:22:50.416079Z	info	Directory /host/secondary-bin-dir is not writable, skipping.
2023-06-22T08:22:50.418197Z	info	Cleaning up.
2023-06-22T08:22:50.418336Z	info	Removing binary: /host/opt/cni/bin/v2-1-istio-cni
2023-06-22T08:22:50.427119Z	info	Removing binary: /host/opt/cni/bin/v2-1-istio-iptables
Error: open /host/etc/cni/multus/net.d/v2-1-istio-cni.kubeconfig.tmp.622916984: no such file or directory
Usage:
  install-cni [flags]

Flags:
      --chained-cni-plugin               Whether to install CNI plugin as a chained or standalone (default true)
      --cni-binaries-prefix string       The filename prefix to add to each binary when copying
      --cni-conf-name string             Name of the CNI configuration file
      --cni-net-dir string               Directory on the host where CNI networks are installed (default "/etc/cni/net.d")
      --cni-network-config string        CNI config template as a string
      --cni-network-config-file string   CNI config template as a file
  -h, --help                             help for install-cni
      --kube-ca-file string              CA file for kubeconfig. Defaults to the pod one
      --kubecfg-file-name string         Name of the kubeconfig file (default "ZZZ-istio-cni-kubeconfig")
      --kubeconfig-mode int              File mode of the kubeconfig file (default 384)
      --log-level string                 Fallback value for log level in CNI config file, if not specified in helm template (default "warn")
      --mounted-cni-net-dir string       Directory on the container where CNI networks are installed (default "/host/etc/cni/net.d")
      --skip-cni-binaries stringArray    Binaries that should not be installed
      --skip-tls-verify                  Whether to use insecure TLS in kubeconfig file
      --update-cni-binaries              Update binaries (default true)
```
The same error appears in the install-cni-v2-2 container. Installed is Red Hat OpenShift Service Mesh v2.4.0.

**Comment by luksa** [comment]
@tyxeron Hi. We'll be releasing 2.4.1 very soon with this fix. 

In the meantime, you can use the workaround explained here: https://access.redhat.com/solutions/7020134

**Comment by jwendell** [comment]
@luksa it takes some time after a pr changing the builder image takes effect.
1 - postsubmit job will build and push it to quay
2 - an hourly cron job in openshift ci fetches new images in quay and feeds an internal image stream that's actually used to run the tests.

**Pull request merged**